### ISSUE TITLE
Contents overlap header item

### ISSUE THREAD

**Issue by MWalid** [kanytu/android-parallax-recyclerview#7]
Is it possible to make `RecyclerView` content overlaps little bit upon the header, something like Google's Play Newsstand??

I have tried to mimic that behavior by adding an additional offset to [`setClipY`][1] function, I could make contents overlap header but the problem is the header disappears early when first item reaches top edge of the view port.

Also the header appears with additional offset when scrolling down again.

Demo:
![Demo](http://s21.postimg.org/5lb4y1293/preview.gif)

Code:
```java
public void setClipY(int offset) {
    int tmpOffset = (int)(offset * 0.7);
    mOffset = offset + tmpOffset;
    invalidate();
}
```

  [1]: https://github.com/kanytu/android-parallax-recyclerview/blob/master/src/main/java/com/poliveira/parallaxrecyclerview/adapter/ParallaxRecyclerAdapter.java#L209

**Comment by kanytu** [comment]
I spent my whole morning trying to fix this issue with no success. The problem is that layoutManager is detaching the header from the recyclerview when the card reaches the top. I posted a question on stackoverflow and I will keep looking for a solution to this problem:

http://stackoverflow.com/questions/28092491/recyclerview-detaches-view-still-visible

**Comment by MWalid** [comment]
@kanytu I really appreciate your time & effort trying to solve it, I also spent a whole day trying :smile: and I'll still struggling till it being solved.

**Comment by kanytu** [comment]
So I found a workaround for this. You will have to update your version because I made some changes.

Next you will have to change your activity/fragment layout so that you have a view just like the header **behind** your RecyclerView. Next change your current header to have an alpha of 0. That will make him transparent. This way the header you actually see when you run is not the header but the view you added behind the RecyclerView. Next implement `ParallaxRecyclerAdapter.OnParallaxScroll()` and inside it translate the other header.

I've uploaded a new branch to my example-project so that you see what needs to be changed.
https://github.com/kanytu/example-parallaxrecycler/tree/issue7

Here is the final result
![ezgif com-optimize 6](https://cloud.githubusercontent.com/assets/560921/5873243/47263df8-a2ec-11e4-8dbf-1e592c13ef9a.gif)

I will leave the issue open until someone finds a better solution

**Comment by sirvon** [comment]
is that example displaying a cardview like the above gif?

where can I view the code behind the recyclerview with cards?

thanks

**Comment by kanytu** [comment]
Yes @sirvon. Tha'ts a CardView layout. Refer to this https://developer.android.com/training/material/lists-cards.html for more information.

A simple example of a row layout is:

    <android.support.v7.widget.CardView
        xmlns:card_view="http://schemas.android.com/apk/res-auto"
        xmlns:android="http://schemas.android.com/apk/res/android"
        android:id="@+id/card_view"
        android:layout_gravity="center"
        android:layout_width="match_parent"
        android:layout_height="150dp"
        android:layout_margin="5dp"
        card_view:cardCornerRadius="4dp">
    
            <TextView
                android:id="@+id/info_text"
                android:layout_width="match_parent"
                android:layout_height="match_parent" />
        </android.support.v7.widget.CardView>

**Comment by MWalid** [comment]
@kanytu I really thank you for your effort to make it work :+1:, sorry for late response because I was kinda busy past 3 days.

**Comment by kanytu** [comment]
I've created a branch with an attempt to fix this:

You should use now `HeaderLayoutManagerFixed` instead of `LinearLayoutManager`.

You should also call:

    HeaderLayoutManagerFixed.setHeaderIncrementFixer(header); //where header is your inflated header.

And also call:
    `ParallaxRecyclerAdapter.setShouldClickView(false);` do disable the clipping. You have to call this before calling `setParallaxHeader`.

**Comment by saitejdandge** [comment]
Try, This Library .Helped me !


https://github.com/florent37/MaterialViewPager
